examples/src/bin/fingerprint-resist.rs

```diff
@@ -0,0 +1,174 @@
+//! This is an example to show how to modify ClientHello to resist
+//! TLS fingerprinting techniques.
+
+use std::io::{stdout, Read, Write};
+use std::net::TcpStream;
+use std::sync::Arc;
+
+use rustls::client::ClientHelloCamouflager;
+use rustls::internal::msgs::base::Payload;
+use rustls::internal::msgs::enums::{ECPointFormat, ExtensionType, PSKKeyExchangeMode};
+use rustls::internal::msgs::handshake::ClientExtension::*;
+use rustls::internal::msgs::handshake::{
+    ClientExtension, ClientSessionTicket, ProtocolName, UnknownExtension,
+};
+use rustls::CipherSuite::*;
+use rustls::{CipherSuite, NamedGroup, ProtocolVersion};
+use rustls::{RootCertStore, SignatureScheme};
+
+fn main() {
+    let mut root_store = RootCertStore::empty();
+    root_store.extend(
+        webpki_roots::TLS_SERVER_ROOTS
+            .iter()
+            .cloned(),
+    );
+    let mut config = rustls::ClientConfig::builder()
+        .with_safe_defaults()
+        .with_root_certificates(root_store)
+        .with_no_client_auth();
+
+    config.alpn_protocols = vec!["http/1.1".as_bytes().to_vec()];
+
+    struct Chrome102;
+    impl ClientHelloCamouflager for Chrome102 {
+        fn transform_cipher_suites(&self, _cipher_suites: Vec<CipherSuite>) -> Vec<CipherSuite> {
+            vec![
```

Comment:
This example should probably at least take the input into account somehow.

Reply:
In this particular example, the goal is to masquerade as a Chrome browser, so it's unnecessary to use existing cipher suites. The interface is designed in this form because there are other legitimate use cases like inserting a GREASE cipher while keeping the others same for testing purposes. Nevertheless, a test has been added to actually use the interface.